psydac/api/tests/test_assembly.py

```diff
@@ -56,29 +74,229 @@ def test_field_and_constant(backend):
 
     # Test matrix A
     x = fh.coeffs
-    assert abs(A.dot(x).dot(x) - 1) < 1e-12
+    #TODO change res into np.conj(res) when the conjugate is applied in the dot product in sympde
+    assert abs(x.dot(A.dot(x)) - res) < 1e-12
 
     # Test vector b
-    assert abs(b.toarray().sum() - 1) < 1e-12
+    assert abs(b.toarray().sum() - res) < 1e-12
+    print("PASSED")
+
+#==============================================================================
+def test_bilinearForm_complex(backend, dtype='complex'):
+
```

Comment:
If this tests bilinear forms defined on function spaces with complex codomain, shouldn't `dtype` always be `'complex'`?

Reply:
Yes, we should always have `dtype=='complex'` for this test. When we put `dtype='complex'` in the test function definition, we claim that this test function have the `dtype` variable fixed to the value`'complex'`.

Reply:
If in this test `dtype` is always `'complex'`, then I would not put it as an argument, but as a variable with a fixed value within the function. Otherwise this could cause future bugs, like eventually somehow passing `'float'` in here and the test failing... Since it is only used in line 94, I would just do:

```
def test_bilinearForm_complex(backend):
    kwargs = {'backend': PSYDAC_BACKENDS[backend]} if backend else {}
    domain = Square()
    V = ScalarFunctionSpace('V', domain)
    V.codomain_type = 'complex'
```

Reply:
[160f283](https://github.com/pyccel/psydac/pull/302/commits/160f2832f045576f3eec91613e85ee45a9b7c342)

This commit should correct it.